c7n/filters/waf.py

```diff
@@ -0,0 +1,315 @@
+                    value: empty
+
+    """
+
+    cache_key = 'c7n:WebACL'
+    associated_cache_key = 'c7n:AssociatedResources'
+
+    schema = type_schema(
+        'wafv2-enabled',
+        rinherit=ValueFilter.schema,
+        **{
+            'web-acl': {'type': 'string'},
+            'state': {'type': 'boolean'}
+        }
+    )
+
+    permissions = (
+        'wafv2:ListWebACLs',
+        'wafv2:GetWebACL', # for augment
+        'wafv2:ListResourcesForWebACL' # for finding associated regional resources
+    )
+
+    def __init__(self, data, manager=None):
+        super().__init__(data, manager)
+
+        # "legacy" mode matches previous WAF based filters for backwards compatability and is
+        # enabled when one of the legacy properties is provided or when no value filter properties
+        # are given (none of the legacy properties are required)
+        self._is_legacy = (
+            'web-acl' in self.data
+                or 'state' in self.data
+                or len(self.data.keys()) == 1 # only filter "type" is given
+        )
+        self._cached_web_acls = None
+
+    def _legacy_match(self, r_acl):
+        target_acl = self.data.get('web-acl')
+        state = self.data.get('state', False)
+
+        return (
+            bool(r_acl)
+            and (
+                target_acl is None
+                or bool(re.match(target_acl, r_acl['Name']))
+            )
+        ) == state
+
+    # get the set of web acls we should look through by asking the resource manager for the set
+    # based on the scope
+    def _get_web_acls(self, scope):
+        if self._cached_web_acls is None:
+            self._cached_web_acls = self.manager.get_resource_manager('wafv2').resources(
+                query=dict(Scope=scope),
+                # required to get the additional detail needed for this filter (e.g. Rules), but
+                # the legacy mode does not require additional detail
+                augment=(not self._is_legacy)
+            )
+
+        return self._cached_web_acls
+
+    # simple search over the list of web acls matching on the specified attribute, returns
+    # None if no match
+    def _get_associated_web_acl_by_attr(self, attr_name, attr_value, scope):
+        web_acls = self._get_web_acls(scope)
+
+        return next(
+            filter(lambda acl: acl[attr_name] == attr_value, web_acls),
+            # default empty so we can actually match where no web acl is present
+            {}
+        )
+
+    # load the resources the web_acl is attached to and cache them with the web acl
+    # we only need to do this for REGIONAL web acls as cloudfront holds a reference to
+    # web acl
+    def _load_associated_resources(self, web_acl, resource_type):
+        cache_key = f'{self.associated_cache_key}:{resource_type}'
+
+        if cache_key not in web_acl:
+            client = self.manager.session_factory().client('wafv2')
+
+            web_acl[cache_key] = client.list_resources_for_web_acl(
```

Comment:
might need pagination

Reply:
Looks like pagination isn't supported for this operation (as in there's no defined paginator, and the API also returns no tokens or hints regarding truncated responses)